src/plans/solver_state.jl

```diff
@@ -211,58 +211,104 @@ Initialize the Functor to a set of keys, where the dictionary is initialized to
 be empty. Further, `once` determines whether more that one update per iteration
 are effective, otherwise only the first update is stored, all others are ignored.
 """
-mutable struct StoreStateAction <: AbstractStateAction
-    values::Dict{Symbol,<:Any}
-    keys::NTuple{N,Symbol} where {N}
+mutable struct StoreStateAction{TPS,TXS,TPI,TTI} <: AbstractStateAction
+    values::Dict{Symbol,Any}
+    keys::Vector{Symbol} # for values
```

Comment:
nice idea to map them to fields of the state :) Not yet sure why a vector is enough, will have to check the code closer.

Reply:
Hm, I thought that was your idea? I've just changed it from tuple to vector to make it type stable and easier to use :slightly_smiling_face: .

Reply:
I am not 100% sure my implementation was the best achievable one. since the symbols are anyways stored in `keys(values)` I am not sure we actually need `keys` here?

Reply:
(which is more reflecting on my choices back long long time ago then your current rework, sure).

Reply:
Separate keys are useful for differentiating "things that are stored" from "things we want to store but we don't yet" so I've left them separate. Basically what my `init` fields are doing for points and tangent vectors.

Reply:
Ah! Then I also understand your `Init`s (and was maybe confused that I could not see their types).